schema/reportdb/reportdb-user-management.sh

```diff
@@ -0,0 +1,176 @@
+#! /bin/bash
+
+DBNAME="reportdb"
+INTERACTIVE=1
+INTERACTIVE_RETRIES=3
+
+if [ 0$UID -gt 0 ]; then
+    echo Run as root.
+    exit 1
+fi
+
+
+print_help() {
+    cat <<HELP
+usage: reportdb-user-management.sh [options]
+
+options:
+  --help 
+            show this help message and exit
+  --non-interactive
+            Switches to non-interactive mode
+  --dbuser=DBUSER
+            Report DB User
+  --dbpassword=DBPASSWORD
+            Report DB Password
+  --add
+            Add the new user
+  --delete
+            Delete the user
+  --modify
+            Set a new password
+
+HELP
+    exit 1
+}
+
+default_or_input() {
+    local MSG="$1"
+    local VARIABLE="$2"
+    local DEFAULT="$3"
+    local SILENT="$4"
+
+    local INPUT
+    local CURRENT_VALUE=${!VARIABLE}
+    #in following code is used not so common expansion
+    #var_a=${var_b:-word}
+    #which is like: var_a = $var_b ? word
+    DEFAULT=${CURRENT_VALUE:-$DEFAULT}
+    local VARIABLE_ISSET=$(set | grep "^$VARIABLE=")
+
+    echo -n "$MSG [$DEFAULT]: "
+    if [ "$INTERACTIVE" = "1" ] &&  [ -z "$VARIABLE_ISSET" ]; then
+        if [ "$SILENT" = 1 ]; then
+          read -sr INPUT
+        else
+          read -r INPUT
+        fi
+    elif [ -z "$VARIABLE_ISSET" ]; then
+        echo "$DEFAULT"
+    else
+        DEFAULT=${!VARIABLE}
+        echo "$DEFAULT"
+    fi
+    if [ -z "$INPUT" ]; then
+        if [ "$DEFAULT" = "y/N" ] || [ "$DEFAULT" = "Y/n" ]; then
+            INPUT=$(yes_no "$DEFAULT")
+        else
+            INPUT="$DEFAULT"
+        fi
+    fi
+    ACCUMULATED_ANSWERS+=$(printf "\n%q=%q" "$VARIABLE" "${INPUT:-$DEFAULT}")
+    eval "$(printf "%q=%q" "$VARIABLE" "$INPUT")"
+}
+
+set_value() {
+    local OPTION="$1"
+    local VAR="$2"
+    local ARG="$3"
+    [[ "$ARG" =~ ^- ]] \
+        && echo "$0: option $OPTION requires argument! Use answer file if your argument starts with '-'." \
+        && print_help
+    eval "$(printf "%q=%q" "$VAR" "$ARG")"
+}
+
+OPTS=$(getopt --longoptions=help,non-interactive,add,delete,modify,action:,dbuser:,dbpassword: -n '${0##*/}' -- h "$@")
```

Comment:
```suggestion
OPTS=$(getopt --longoptions=help,non-interactive,add,delete,modify,action:,dbuser:,dbpassword: -n "${0##*/}" -- h "$@")
```

You need double quotes here, otherwise the substitution won't happen.